master/internal/provisioner/aws.go

```diff
@@ -161,7 +176,47 @@ func (c *awsCluster) dryRunRequests() error {
 	return nil
 }
 
+func (c *awsCluster) setup(ctx *actor.Context) {
```

Comment:
I saw the function is called `attemptToApproximateClockSkew`. Is it possible it might fail? Also, this function can actually be moved to `newAWSCluster`.

Reply:
It could fail to approximate the clock skew*, but the setup process can't fail in a fatal way. It shouldn't be moved to newAWSCluster because that function is executed serially during the creation of the `resourcePool`. We shouldn't block the master just for this setup step.

* this could lead to a bad UX for users who have really large clock skew or users who have their clock ahead of AWS's